### DISCUSSION TITLE
.json file missing during initial audio processing

### DISCUSSION THREAD

**Discussion by cheriestoner** [dmarx/video-killed-the-radio-star#157] [Q&A]
Run all previous cells as instructed, and error occurs as:

Transcribing...
Detecting language using up to the first 30 seconds. Use `--language` to specify the language
Detected language: Chinese
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/whisper/transcribe.py", line 478, in cli
    result = transcribe(model, audio_path, temperature=temperature, **args)
  File "/usr/local/lib/python3.10/dist-packages/whisper/transcribe.py", line 323, in transcribe
    add_word_timestamps(
  File "/usr/local/lib/python3.10/dist-packages/whisper/timing.py", line 298, in add_word_timestamps
    alignment = find_alignment(model, tokenizer, text_tokens, mel, num_frames, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/whisper/timing.py", line 210, in find_alignment
    weights = median_filter(weights, medfilt_width)
  File "/usr/local/lib/python3.10/dist-packages/whisper/timing.py", line 40, in median_filter
    result = median_filter_cuda(x, filter_width)
  File "/usr/local/lib/python3.10/dist-packages/whisper/triton_ops.py", line 107, in median_filter_cuda
    kernel[(grid,)](y, x, x.stride(-2), y.stride(-2), BLOCK_SIZE=BLOCK_SIZE)
  File "<string>", line 63, in kernel
  File "/usr/local/lib/python3.10/dist-packages/triton/compiler/compiler.py", line 425, in compile
    so_path = make_stub(name, signature, constants)
  File "/usr/local/lib/python3.10/dist-packages/triton/compiler/make_launcher.py", line 39, in make_stub
    so = _build(name, src_path, tmpdir)
  File "/usr/local/lib/python3.10/dist-packages/triton/common/build.py", line 61, in _build
    cuda_lib_dirs = libcuda_dirs()
  File "/usr/local/lib/python3.10/dist-packages/triton/common/build.py", line 30, in libcuda_dirs
    assert any(os.path.exists(os.path.join(path, 'libcuda.so')) for path in dirs), msg
AssertionError: libcuda.so cannot found!

Skipping /content/drive/MyDrive/AI/VideoKilledTheRadioStar/shared_assets/DOWNLOADED__给你一瓶魔法药水 You are My Magic- 告五人 Accusefive 【动态歌词 lyric video】with English Subtitles.m4a due to AssertionError: libcuda.so cannot found!

---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
[<ipython-input-10-7f62cc2c8dbe>](https://localhost:8080/#) in <cell line: 38>()
     46     audio_meta['duration'] = get_audio_duration_seconds(audio_fpath)
     47 
---> 48     with whisper_seg_fpath.open() as f:
     49         timings = json.load(f)
     50     whisper_seg = timings['segments']

[/usr/lib/python3.10/pathlib.py](https://localhost:8080/#) in open(self, mode, buffering, encoding, errors, newline)
   1117         if "b" not in mode:
   1118             encoding = io.text_encoding(encoding)
-> 1119         return self._accessor.open(self, mode, buffering, encoding, errors,
   1120                                    newline)
   1121 

FileNotFoundError: [Errno 2] No such file or directory: '/content/drive/MyDrive/AI/VideoKilledTheRadioStar/shared_assets/DOWNLOADED__给你一瓶魔法药水 You are My Magic- 告五人 Accusefive 【动态歌词 lyric video】with English Subtitles.json'

**Comment by npgilbertson** [comment]
**Seeing as this appears dead and not had this simple update in months**

I havent seen if the rest works but so far 
_Replace the code in "Install dependencies with:_
```
 #%%capture

# @title # 🛠️ Setup: Install Dependencies

# Install dependencies

try:
    import google.colab
    local=False
except:
    local=True

# TODO: pin versions

# local only additional dependencies
if local:
    %pip install pandas torch pillow beautifulsoup4 scipy toolz numpy lxml librosa scikit-learn rich

# dependencies for both colab and local
%pip install torch==2.0.1 triton==2.0.0 yt-dlp python-tsp stability-sdk[anim_ui] diffusers transformers ftfy accelerate omegaconf
%pip install openai-whisper panel huggingface_hub ipywidgets safetensors keyframed demucs parse""
```

Is at least getting me past that first stage. Basically the issue isn't the json file at all. It's before that, in that triton V2.1 doesn't work with another guy in there. Causing cuda.so to not be found. Code tries to continue (should have an error system for anything file related imo to break the code) and tries to find it's not yet made json because it failed earlier. I also made an empty .json file with just 
[
]
in the file, but thats not needed with the above code!
